subsys/bluetooth/host/crypto.c

```diff
@@ -111,6 +136,88 @@ int bt_rand(void *buf, size_t len)
 }
 #endif /* CONFIG_BT_HOST_CRYPTO_PRNG */
 
+#if defined(CONFIG_BT_USE_PSA_API)
+int bt_encrypt_le(const uint8_t key[16], const uint8_t plaintext[16],
+		  uint8_t enc_data[16])
+{
+	psa_key_attributes_t attr = PSA_KEY_ATTRIBUTES_INIT;
+	psa_key_id_t key_id = MBEDTLS_SVC_KEY_ID_INIT;
+	psa_status_t status, destroy_status;
+	size_t out_len;
+	uint8_t tmp[16];
+
+	CHECKIF(key == NULL || plaintext == NULL || enc_data == NULL) {
+		return -EINVAL;
+	}
+
+	LOG_DBG("key %s", bt_hex(key, 16));
+	LOG_DBG("plaintext %s", bt_hex(plaintext, 16));
+
+	sys_memcpy_swap(tmp, key, 16);
+
+	psa_set_key_type(&attr, PSA_KEY_TYPE_AES);
+	psa_set_key_bits(&attr, 128);
+	psa_set_key_usage_flags(&attr, PSA_KEY_USAGE_ENCRYPT);
+	psa_set_key_algorithm(&attr, PSA_ALG_SHA_256);
```

Comment:
Why different crypto algorithms are used in Host with PSA (sha256) and without PSA (aes128)?

Reply:
Ops, sorry, that's definitely an error, of course. I will fix this

Reply:
Question: since I don't have any physical board which supports both bluetooth and TFM, do you think there is a way I can test these changes through emulation? Test cases that I modified in this PR only build tests, which is not super bad, but this is not enough to find errors like the one you just caught with this comment.

`nrf5340dk/nrf5340/cpuapp/ns` should be a good target and since that's also used in `bsim` I thought that this could be a nice way to test. However I didn't find any `bsim` test for the `nrf5340dk` platform which can test the changes I'm introducing with this PR. Am I missing something?

Reply:
@aescolar can probably help you with that, he recently refactored nrf53/54 bsim tests.

Reply:
I'm afraid nrf53 support as bsim emulated chip is on ready yet to full extent. It doesn't support TrustZone, does it?
Nevertheless, I'm pretty sure, it is possible to use\run Host's bsim tests with `CONFIG_BT_USE_PSA_API=y`.
If PSA was added correctly over KConfig\Cmake they should be built without problem (excluding ITS that is not thread safe in mbedTLS). We added this ability into mesh bsim tests. I do not think there is big difference between mesh and hosts tests.

Reply:
Thanks a lot for the suggestion!! I added a new commit (the last one - https://github.com/zephyrproject-rtos/zephyr/pull/73378/commits/cb6f66d2ec636029928e1ef044fce6553a743ac9) to add a bsim test case for `CONFIG_BT_USE_PSA_API` for the nrf52840dk platform.